cache/src/main/scala/coursier/FileError.scala

```diff
@@ -16,6 +16,11 @@ sealed abstract class FileError(
 
 object FileError {
 
+  final case class RetryExhausted(reason: String) extends FileError(
```

Comment:
Maybe return the last `FileError` instead of this one? Or keep the last error as a field of that one?

Reply:
Used the former and logs "Retry exhausted..." in console.